pages/doc.js

```diff
@@ -32,59 +32,69 @@ export default class Documentation extends Component {
       markdown: '',
       headings: [],
       pageNotFound: false,
-      isMenuOpen: false
+      isMenuOpen: false,
+      load: false
     }
   }
-
   componentDidMount() {
     this.loadStateFromURL()
     this.initDocsearch()
     window.addEventListener('popstate', this.loadStateFromURL)
-    this.ps = new PerfectScrollbar('#sidebar-menu', {
-      // wheelPropagation: window.innerWidth <= 572
-      wheelPropagation: true
-    })
-  }
-
-  componentDidUpdate() {
-    this.ps.update()
   }
-
   componentWillUnmount() {
     window.removeEventListener('popstate', this.loadStateFromURL)
   }
-
-  loadStateFromURL = () => {
-    const { pathname } = window.location
-    const sectionURL = pathname.split('/')[2] // match section from URL
-    const sectionIndex = sidebar.findIndex(
-      section => (section.slug || kebabCase(section.name)) === sectionURL
-    )
-    if (sectionIndex === -1) {
-      sectionURL
-        ? this.setState({ pageNotFound: true })
-        : this.onSectionSelect(0)
+  toString(method, str) {
+    switch (method) {
+      case 'filetourl':
+        return str.slice(0, -3)
+        break
+      default:
+        break
+    }
+  }
+  getZeroFile(arr) {
+    if (typeof arr[0] !== 'string' && arr[0].indexFile) {
+      return arr[0].indexFile
+    } else if (typeof arr[0] !== 'string') {
+      return arr[0]
     } else {
-      const fileURL = pathname.split('/')[3] // match file from URL
-      const sectionFiles = flatten(sidebar[sectionIndex].files)
-      const fileIndex = sectionFiles.findIndex(
-        file => kebabCase(file.slice(0, -3)) === fileURL
-      )
-      if (fileIndex === -1) {
-        fileURL
-          ? this.setState({ pageNotFound: true })
-          : this.onSectionSelect(sectionIndex)
-      } else {
-        this.loadFile({
-          section: sectionIndex,
-          file: sectionFiles[fileIndex],
-          parseHeadings: true,
-          pageNotFound: false
-        })
+      this.getZeroFile(arr[0].files)
+    }
+  }
+  loadStateFromURL = () => {
+    let file = this.getZeroFile(sidebar) //установили начальный файл
+    let indexes = []
+    let path = window.location.pathname.split('/')
+    let length = path.length
+    function getFile(arr, find, x) {
+      for (let i = 0; i < arr.length; i++) {
+        if (
+          (typeof arr[i] === 'string' && arr[i].slice(0, -3) === find) ||
+          (arr[i].indexFile && arr[i].indexFile.slice(0, -3) === find)
+        ) {
+          file = arr[i]
+          indexes.push(i)
+          return false
+        } else if (arr[i].name && kebabCase(arr[i].name) === find) {
+          indexes.push(i)
+          file = arr[i].files[0]
+          return false
+        } else if (arr[i].files) {
+          getFile(arr[i].files, find, x)
+        }
       }
     }
+    for (let x = 2; x < length; x++) {
```

Comment:
just remove the prefix at the very beginning, this way harder to understand

Reply:
ok done